src/utils/rocksdb_serialization.hpp

```diff
@@ -30,6 +30,50 @@ namespace memgraph::utils {
 static constexpr const char *outEdgeDirection = "0";
 static constexpr const char *inEdgeDirection = "1";
 
+namespace {
+struct StartEndPositions {
+  size_t start;
+  size_t end;
+
+  size_t Size() const { return end - start; }
+  bool Valid() const { return start != std::string::npos; }
+};
+
+inline StartEndPositions FindPartStartEndPositions(const std::string_view str, const char delim,
+                                                   unsigned int partNumber) {
+  StartEndPositions result{0, 0};
+  for (int i = 0; i < partNumber; ++i) {
+    result.start = result.end;
+    result.end = str.find(delim, result.start);
+    if (result.end == std::string::npos) {
+      if (i < partNumber - 1) {
+        // We didn't find enough parts.
+        result.start = std::string::npos;
+      }
+      return result;
+    }
+    ++result.end;
+  }
+  return result;
+}
+
+inline std::string_view GetViewOfFirstPartOfSplit(const std::string_view src, const char delimiter) {
+  size_t delimPos = src.find(delimiter);
```

Comment:
Could we just use the FindPartStartEndPositions here as well, or is the penalty too high?